Sources/_StringProcessing/Regex/Core.swift

```diff
@@ -68,20 +68,35 @@ extension Regex {
   }
 }
 
+
 @available(SwiftStdlib 5.7, *)
 extension Regex {
   /// A program representation that caches any lowered representation for
   /// execution.
-  internal class Program {
+  internal final class Program: @unchecked Sendable {
     /// The underlying IR.
     ///
     /// FIXME: If Regex is the unit of composition, then it should be a Node instead,
     /// and we should have a separate type that handled both global options and,
     /// likely, compilation/caching.
     let tree: DSLTree
 
+    private final class ProgramBox {
```

Comment:
Why do we need two boxes?

Reply:
`UnsafeAtomicLazyReference` is generic over an `AnyObject`. I guess we could skip that type and just store this as a `Optional<UnsafeMutablePointer<MEProgram<String>>>` — that would save one level of indirection, right?

Reply:
Updated implementation only uses one layer of indirection 👍🏻